temporian/implementation/numpy_cc/operators/calendar.cc

```diff
@@ -0,0 +1,258 @@
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+#include "absl/time/time.h"
+
+#include "temporian/implementation/numpy_cc/operators/common.h"
+
+namespace {
+namespace py = pybind11;
+
+static absl::TimeZone parse_tz(const py::object tz) {
+  if (py::isinstance<py::int_>(tz)) {
+    // handle tz as an int
+    const int int_tz = tz.cast<int>();
+    // tz is expresed in hours, needs to be converted to seconds
+    return absl::FixedTimeZone(int_tz * 60 * 60);
+
+  } else if (py::isinstance<py::float_>(tz)) {
+    // handle tz as an float
+    const float float_tz = tz.cast<float>() * 60 * 60;
+    const int int_tz = static_cast<int>(std::round(float_tz));
+    // tz is expresed in hours, needs to be converted to seconds
+    return absl::FixedTimeZone(int_tz);
+
+  } else if (py::isinstance<py::str>(tz)) {
+    // handle tz as an string
+    const std::string str_tz = tz.cast<std::string>();
+    absl::TimeZone parsed_tz;
+    if (!absl::LoadTimeZone(str_tz, &parsed_tz)) {
+      throw std::invalid_argument("Invalid timezone '" + str_tz +
```

Comment:
Blocker

Reply:
Changed it to avoid raising exceptions. the new implementation returns an error message